### ISSUE TITLE
web3.js: CI tests failing

### ISSUE THREAD

**Issue by jordansexton** [solana-labs/solana#23615]
#### Problem

Web3.js CI tests are failing in master, which is seen on unrelated PRs e.g. https://github.com/solana-labs/solana/pull/23579

@jacobcreech on [Discord](https://discord.com/channels/428295358100013066/476811830145318912/951886952372772929):
> There's a bunch of npm test failures on test:live with similar messages to Error: failed to get confirmed block: Block 1 cleaned up, does not exist on node. First available block: 2
> They're failing with these on the local validator from the looks of it

#### Proposed Solution

> There needs to be some error handling from the looks of it on this while loop in the tests https://github.com/solana-labs/solana/blob/master/web3.js/test/connection.test.ts#L1700-L1708
> That or longer storage enabled on the local validator

**Comment by steveluscher** [comment]
I've found, by bisecting version-wise, that as of version 1.9.10 of the validator, blocks 0 and 1 disappear like this. These tests last passed on 1.9.9.

We're discussing what to do here: https://discord.com/channels/428295358100013066/439194979856809985/955874011529764984

cc/ @mvines, @t-nelson.

**Comment by steveluscher** [comment]
Looks like the curious case of the disappearing blocks started with https://github.com/solana-labs/solana/pull/23403.

**Comment by steveluscher** [comment]
Boom! Other than the tractable, actual Mocha test failures that have been fixed by #23921, I figured out why the test validator crashes in CI. Check it out.

```
[2022-03-25T03:49:18.616695138Z ERROR solana_ledger::blockstore] Unable to increase the maximum open file descriptor limit to 5
thread 'main' panicked at 'Failed to open ledger database: UnableToSetOpenFileDescriptorLimit', core/src/validator.rs:1304:6stack backtrace:
   0:     0x562adaf8b75c - std::backtrace_rs::backtrace::libunwind::trace::h91c465e73bf6c785                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/../../backtrace/src/backtracenwind.rs:93:5
   1:     0x562adaf8b75c - std::backtrace_rs::backtrace::trace_unsynchronized::hae9da36f5d58b5f3
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/../../backtrace/src/backtracers:66:5
   2:     0x562adaf8b75c - std::sys_common::backtrace::_print_fmt::h7f499fa126a7effb
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x562adaf8b75c - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h3e2b509ce2ce6007
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x562adafb715c - core::fmt::write::h753c7571fa063ecb
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/fmt/mod.rs:1168:17
   5:     0x562adaf80bc3 - std::io::Write::write_fmt::h2815c0519c99ba09
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/io/mod.rs:1660:15
   6:     0x562adaf8e0d2 - std::sys_common::backtrace::_print::h64941a6fc8b0ed9b
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x562adaf8e0d2 - std::sys_common::backtrace::print::hcf25e43e1a9b0766
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x562adaf8e0d2 - std::panicking::default_hook::{{closure}}::h78d3e6cf97fc623d
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:211:50
   9:     0x562adaf8dcb5 - std::panicking::default_hook::hda898f8d3ad1a5ae
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:228:9
  10:     0x562adaf8e723 - std::panicking::rust_panic_with_hook::h1a5ea2d6c23051aa
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:606:17
  11:     0x562adaf8e440 - std::panicking::begin_panic_handler::{{closure}}::h07f549390938b73f
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:502:13
  12:     0x562adaf8bc04 - std::sys_common::backtrace::__rust_end_short_backtrace::h5ec3758a92cfb00d
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/sys_common/backtrace.rs:139:1
  13:     0x562adaf8e179 - rust_begin_unwind
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
  14:     0x562ad94b70d1 - core::panicking::panic_fmt::h3a79a6a99affe1d5
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
  15:     0x562ad94b71c3 - core::result::unwrap_failed::ha0327e3803285d6e
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/result.rs:1690:5
  16:     0x562ad96c58c1 - solana_core::validator::Validator::new::hf3f6e5a3a64f0dd6
  17:     0x562ad95adf95 - solana_test_validator::TestValidator::start::hc50c0b307d03c551
  18:     0x562ad95aaede - solana_test_validator::TestValidatorGenesis::start_with_mint_address::h09b6c371160c1714
  19:     0x562ad95076bb - solana_test_validator::main::h6740ca89ee2248e5
  20:     0x562ad94ee373 - std::sys_common::backtrace::__rust_begin_short_backtrace::h42826ec4b2b58f9e
  21:     0x562ad94f61c9 - std::rt::lang_start::{{closure}}::h7073d7fb3e2b7879
  22:     0x562adaf8ae10 - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h443f738a8e9f94
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ops/function.rs:259:13
  23:     0x562adaf8ae10 - std::panicking::try::do_call::h1e21ba261ba489ec
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:406:40
  24:     0x562adaf8ae10 - std::panicking::try::h6afd48af8b6c96ac
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:370:19
  25:     0x562adaf8ae10 - std::panic::catch_unwind::h85dd95e0bab7fb60
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panic.rs:133:14
  26:     0x562adaf8ae10 - std::rt::lang_start_internal::{{closure}}::h038455e697c8b03e
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/rt.rs:128:48
  27:     0x562adaf8ae10 - std::panicking::try::do_call::h6b0ad65979f3077a
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:406:40
  28:     0x562adaf8ae10 - std::panicking::try::h010108d314169ac6
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:370:19
  29:     0x562adaf8ae10 - std::panic::catch_unwind::hff397f912b1535c2
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panic.rs:133:14
  30:     0x562adaf8ae10 - std::rt::lang_start_internal::h52e73755f77c7dd9
                               at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/rt.rs:128:20
  31:     0x562ad950a192 - main
  32:     0x7f057482a0b3 - __libc_start_main
  33:     0x562ad94d9fba - _start
  34:                0x0 - <unknown>
```

I figured this out by adding an SSH terminal step to the GitHub Action workflow, shelling into the CI server, and trying to start the validator with `RUST_BACKTRACE=full solana-test-validator`.

```
- name: Setup upterm session
  uses: lhotari/action-upterm@v1
  with:
    limit-access-to-actor: true
    limit-access-to-users: steveluscher
```

I'll add a step to raise the max file descriptors in #23923.
